src/garage/tf/policies/categorical_cnn_policy.py

```diff
@@ -101,9 +99,12 @@ def __init__(self,
         self._initialize()
 
     def _initialize(self):
-        state_input = tf.compat.v1.placeholder(tf.float32,
+        state_input = tf.compat.v1.placeholder(tf.uint8,
                                                shape=(None, ) + self.obs_dim)
 
+        if isinstance(self.env_spec.observation_space, akro.Image):
```

Comment:
a CategoricalCNNPolicy can only accept akro.Images, so I think you should remove these conditionals and instead add a guard in the constructor

```py
if not isinstance(env_spec.observation_space, akro.Image):
    raise ValueError('CategoricalCNNPolicy can only process akro.Image observations, but received an env_spec with observation_space of type {}'
    .format(type(env_spec.observation_space).__name__))
```

Reply:
Does this apply to all CNN primitives? I was under the impression that CNNs can be used on non-image data as well.

Reply:
You can apply a CNNs to arbitrary 2D data, which of course doesn't have to be an image (but usually are).

However, for the purposes of type-safety, CNNs operate the type `Image` (i.e. arbitrary 2D data). If, for instance, someone wants their CNNs to operate on sound data, they could rearrange that data into a 2D grid, which we call an image.

Reply:
Shouldn't non-image 2D data be classified as a Box? Using your sound example, if one were to input sound data as an `akro.Image`,  `CategoricalCNNPolicy` will normalize this data by dividing by 255 thinking it contains pixel values.

Reply:
On second thought this might not be an issue since an akro.Image can only contain values between 0 and 255, though it would mean non-image data would have to be mapped to that range as well.

Reply:
How about CNN could take either akro.Image or akro.Box, and only normalize when it's akro.image?

Reply:
I agree @ahtsan. How does that sound @ryanjulian?

Reply:
My preference is to enforce a 1:1 mapping between types and networks (i.e. CNNs take images). The purpose of this change was to simplify the type relationships in this code by eliminating `Box` types, since they are hard to validate, not introduce a new, complicated set of cases.

Perhaps the definition of `akro.Image` is too restrictive, and we should provision for float-valued images or int-valued images, but that adds complexity which also has to be justified by identifying a real use case, not a hypothetical one.

In practical terms, if someone wanted to use a CNN with float-valued data (for instance), converting that data to uint8 has shown to have a small-to-negligible impact on performance in NNs. If this isn't sufficient for them, there's nothing stopping them from writing their own CategoricalCNNPolicy based largely on the existing one.

As for what to actually do, I think accepting `Box` is fine, as long as you validate that it's a 2D or 3D `Box`. The 2D box is assumed to be shaped `(W, H)` and the 3D box is assumed to be shaped `(W, H, C)`.

Reply:
@krzentner i'm curious to hear your thoughts

Reply:
I would also lean towards accepting `Box`, since many people will have environments that don't use the `Image` type, but they would expect garage to handle.

If they chose to use a CNN, their intent is fairly clear anyways.

Reply:
Alright -- so we can merge this when all the CNN types validate that the input is either a 2D/3D Box or an Image.

Reply:
@ryanjulian sounds good. I assume they should also normalize images?